### PR TITLE
Added new resource - Site External EPG

### PR DESCRIPTION
**Pull request by anvitha-jain**
solves #11

### CODE REVIEW

**Review by shrsr — COMMENTED** [review]
Can you use the go linter on this for proper formatting?

File: examples/schema_site_external_epg/main.tf [hunk 1]
```diff
@@ -11,27 +11,106 @@ provider "mso" {
...
-  ip = "12.25.70.50"
-}
\ No newline at end of file
+	schema_id             = mso_schema.schema1.id
+	template_name         = mso_schema.schema1.template_name
+	external_epg_name     = "temp_epg"
+	display_name          = "temp_epg"
+	vrf_name              = mso_schema_template_vrf.vrf1.id
+  vrf_schema_id         = mso_schema_template_vrf.vrf1.schema_id
+  vrf_template_name     = mso_schema_template_vrf.vrf1.template
```
Comment (lhercot) [inline examples/schema_site_external_epg/main.tf:68]: Can you align those with the other lines or use terraform fmt (https://www.terraform.io/docs/cli/commands/fmt.html) to do it for you.

File: examples/schema_site_external_epg/main.tf [hunk 2]
```diff
@@ -11,27 +11,106 @@ provider "mso" {
...
+      "vrfRef": {
+          "schemaId": "${mso_schema_site_vrf.site_vrf.schema_id}",
+          "templateName": "${mso_schema_site_vrf.site_vrf.template_name}",
+          "vrfName": "${mso_schema_site_vrf.site_vrf.vrf_name}"
+        }
+      }
+    }
+  ]
+  EOF
+}
```
Comment (lhercot) [inline examples/schema_site_external_epg/main.tf:107]: Can you open an enhancement to keep track of the site L3Out?

File: examples/schema_site_external_epg/main.tf [hunk 3]
```diff
@@ -11,27 +11,106 @@ provider "mso" {
...
+
+resource "mso_schema_site_external_epg" "site_externalepg" {
+  depends_on            = [ mso_rest.site_l3out ]
+  site_id               = mso_schema_site.schema_site_1.site_id
+	schema_id             = mso_schema_site.schema_site_1.schema_id
+	template_name         = mso_schema_template_external_epg.template_externalepg.template_name
+	external_epg_name     = mso_schema_template_external_epg.template_externalepg.external_epg_name
+  l3out_name            = mso_schema_template_l3out.template_l3out.l3out_name
+}
```
Comment (lhercot) [inline examples/schema_site_external_epg/main.tf:116]: Same formatting issue as previous one.

File: mso/resource_mso_schema_site_external_epg.go [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// Gets tenant name by doing the following
+// GET and loop through all the schemas and check if you schema is present ("api/v1/schemas/list-identity")
```
Comment (lhercot) [inline mso/resource_mso_schema_site_external_epg.go:344]: you schema -> the schema

File: mso/resource_mso_schema_site_external_epg.go [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+// GET and loop through all the templates in your schema and check if your template is present
```
Comment (lhercot) [inline mso/resource_mso_schema_site_external_epg.go:345]: your schema -> the schema
your template  -> the template

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**